"not enough arguments in call to p.client.ContainerCreate"

Installing Testcontainers v0.9.0 currently fails to work, because Go is automatically pulling in Docker v20.10.1 instead of v17.12.0 and the API has changed.

As such, when I run anything I get:
```
/Users/sazzer/.gvm/pkgsets/go1.15.6/global/pkg/mod/github.com/testcontainers/testcontainers-go@v0.9.0/docker.go:645:39: not enough arguments in call to p.client.ContainerCreate
	have (context.Context, *container.Config, *container.HostConfig, *network.NetworkingConfig, string)
	want (context.Context, *container.Config, *container.HostConfig, *network.NetworkingConfig, *v1.Platform, string)
```

I can fix this by running:
```
$ go mod edit -require github.com/docker/docker@v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible
```

But the next time I run `go get -u ./...` to update dependencies then it all breaks again.

ok so i tried earlier versions too didnt work:
`$ go get -u github.com/testcontainers/testcontainers-go@v0.8.0`

output:
```
go: finding github.com/testcontainers/testcontainers-go v0.8.0
go: downloading github.com/testcontainers/testcontainers-go v0.8.0
go: extracting github.com/testcontainers/testcontainers-go v0.8.0
go: finding google.golang.org/genproto latest
go: finding golang.org/x/sync latest
go: finding golang.org/x/sys latest
go: finding github.com/containerd/continuity latest
go: finding golang.org/x/net latest
go: finding github.com/golang/groupcache latest
go: finding github.com/containerd/cgroups latest
# github.com/testcontainers/testcontainers-go
../go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.8.0/docker.go:614:39: not enough arguments in call to p.client.ContainerCreate
        have (context.Context, *container.Config, *container.HostConfig, *network.NetworkingConfig, string)
        want (context.Context, *container.Config, *container.HostConfig, *network.NetworkingConfig, *v1.Platform, string)
```

--- a/docker.go
+++ b/docker.go
@@ -645,7 +645,7 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 		EndpointsConfig: endpointConfigs,
 	}
 
-	resp, err := p.client.ContainerCreate(ctx, dockerInput, hostConfig, &networkingConfig, req.Name)
+	resp, err := p.client.ContainerCreate(ctx, dockerInput, hostConfig, &networkingConfig, nil, req.Name)
 	if err != nil {
 		return nil, err
 	}

--- a/go.mod
+++ b/go.mod
@@ -10,7 +10,7 @@ require (
 	github.com/containerd/containerd v1.4.1 // indirect
 	github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc // indirect
 	github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible // indirect
-	github.com/docker/docker v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible
+	github.com/docker/docker v20.10.3+incompatible
 	github.com/docker/go-connections v0.4.0
 	github.com/docker/go-units v0.3.3 // indirect
 	github.com/go-redis/redis v6.15.9+incompatible
@@ -21,16 +21,17 @@ require (
 	github.com/gorilla/context v1.1.1 // indirect
 	github.com/gorilla/mux v1.6.2 // indirect
 	github.com/kr/pretty v0.1.0 // indirect
+	github.com/moby/sys/mount v0.2.0 // indirect
 	github.com/morikuni/aec v0.0.0-20170113033406-39771216ff4c // indirect
 	github.com/onsi/ginkgo v1.8.0 // indirect
 	github.com/onsi/gomega v1.5.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
-	github.com/opencontainers/image-spec v1.0.1 // indirect
+	github.com/opencontainers/image-spec v1.0.1
 	github.com/opencontainers/runc v0.1.1 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.2.0 // indirect
 	github.com/stretchr/testify v1.7.0
-	golang.org/x/sys v0.0.0-20200116001909-b77594299b42
+	golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c
 	golang.org/x/text v0.3.2 // indirect
 	golang.org/x/time v0.0.0-20181108054448-85acf8d2951c // indirect
 	google.golang.org/grpc v1.17.0 // indirect

